chunks/chunks_test.go

```diff
@@ -26,3 +28,34 @@ func TestReaderWithInvalidBuffer(t *testing.T) {
 	_, err := r.Chunk(0)
 	testutil.NotOk(t, err)
 }
+
+func TestNewReaderWithInvalidChunkHeader(t *testing.T) {
```

Comment:
Where it makes sense I am vouching for [behaviour tests](https://www.youtube.com/watch?v=EZ05e7EMOLM)(using only public API interfaces) and here this would mean to test that reading corrupted chunks from the disk return the correct errors. So it looks like we should use `OpenBlock`.

you can create a block (`createBlock`) , than manually corrupt a chunk and try to open the block. You can have a look at `TestWAL_Repair` for ideas how to corrupt a chunk.

Reply:
Great suggestions.

Thank you!